src/sentry/issues/ingest.py

```diff
@@ -13,6 +42,150 @@ def save_issue_occurrence(occurrence_data: IssueOccurrenceData, event: Event) ->
     # sure that this is somehow validated.
     occurrence.save(event.project_id)
 
-    # TODO: Create/update issue
+    # TODO: Pass release here
+    save_issue_from_occurrence(occurrence, event, None)
+    # TODO: Create group related releases here
     # TODO: Write occurrence and event eventstream
     return occurrence
+
+
+def process_occurrence_data(occurrence_data: IssueOccurrenceData) -> None:
+    # Hash fingerprints to make sure they're a consistent length
+    occurrence_data["fingerprint"] = [
+        md5(part.encode("utf-8")).hexdigest() for part in occurrence_data["fingerprint"]
+    ]
+
+
+class IssueArgs(TypedDict):
+    platform: Optional[str]
+    message: str
+    level: Optional[int]
+    culprit: str
+    last_seen: datetime
+    first_seen: datetime
+    active_at: datetime
+    type: int
+    data: OccurrenceMetadata
+    first_release: Optional[Release]
```

Comment:
at this point we might want all the types to live in their own files

Reply:
This type is specific to this one file., I don't think we need to move it out really. Tbh this specific type might be overkill, although it seems nicer than just using `Mapping[str, Any]`

Reply:
makes sense. i just feel like we're at 3+ TypedDicts for this stuff now haha between occurrences and events and whatnot